### PR TITLE
Fixes for xml.search service

### PR DESCRIPTION
**Pull request by josegar74**
Error reported with http://localhost:8080/geonetwork/srv/eng/xml.search service:

```
"error": { "message": "Failed to compile stylesheet. 37 errors detected." , "class": "TransformerConfigurationException" , "request": {} } }
```

### CODE REVIEW

**Review by juanluisrp — APPROVED** [review]
Tested. LGTM.

File: src/main/plugin/iso19139.ca.HNAP/present/metadata.xsl [hunk 1]
```diff
@@ -15,14 +15,15 @@
...
+                extension-element-prefixes="saxon"
+                exclude-result-prefixes="gmx xsi gmd gco gml gts srv xlink exslt geonet rdf ns2 rdfs skos svrl saxon">
 
   <xsl:import href="metadata-brief.xsl"/>
   <xsl:import href="metadata-fop.xsl"/>
   <xsl:import href="metadata-utils.xsl"/>
-  <xsl:include href="iso19139.ec.HNAP-readonly.xsl"/>
 
-  <!-- main template - the way into processing iso19139.ca.HNAP -->
+  <!-- main template - the way into processing iso19139.nap -->
```
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/present/metadata.xsl]: Is there a reason the comment was changed? It seems like the older one was correct.

File: src/main/plugin/iso19139.ca.HNAP/present/metadata-fop.xsl [hunk 2]
```diff
@@ -41,6 +41,26 @@
       </xsl:call-template>
     </xsl:variable>
 
+    <xsl:variable name="webMapServicesProtocols" select="/root/gui/webServiceTypes" />
+
+    <xsl:variable name="isoLanguages" select="/root/gui/isolanguages" />
+
+    <xsl:variable name="port">
+      <xsl:choose>
+        <xsl:when test="/root/gui/env/server/protocol = 'https'"><xsl:value-of select="/root/gui/env/server/securePort" /></xsl:when>
```
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/present/metadata-fop.xsl]: Looks like 3.12 GN removed this.  
https://github.com/geonetwork/core-geonetwork/pull/5569

It this condition something that should only exists in versions prior to 3.12?

Also seems like this logic to get the site url would be cleaner if it used the "getSiteUrl" from the java classes. Seems odd to add all this logic in xslt when the existing logic seems to exists in java.

**Review by ianwallen — COMMENTED** [review]
I noticed that a lot of  `iso19139.ca.HNAP` was changed to `iso19139.nap`  I thought `iso19139.ca.HNAP` was the correct value?  Could you explain why these were all changed?

File: src/main/plugin/iso19139.ca.HNAP/present/metadata-fop.xsl (hunk 2 above)
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/present/metadata-fop.xsl]: @josegar74 can you answer this question please?

(@ianwallen I am not sure your question blocks this PR or not)

Comment (josegar74) [reply src/main/plugin/iso19139.ca.HNAP/present/metadata-fop.xsl]: Updated the code.

**Comment by jodygarnett** [comment]
With this change http://localhost:8080/geonetwork/srv/eng/xml.search reports back:

```xml
<?xml version="1.0" encoding="utf-8"?>
<response from="1" to="0" selected="0" maxPageSize="100">
  <summary count="0" type="local">
    <types />
    <topicCats />
    <inspireThemes />
    <inspireThemesWithAc />
    <inspireThemesURI />
    <keywords />
    <orgNames />
    <createDateYears />
    <formats />
    <spatialRepresentationTypes />
    <maintenanceAndUpdateFrequencies />
    <status />
    <serviceTypes />
    <denominators />
    <resolutions />
  </summary>
</response>
```

**Pull request merged**